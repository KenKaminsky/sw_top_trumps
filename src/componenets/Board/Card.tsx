import React from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';
import { camelCaseToSentenceCase } from '../../shared/helpers';
import { IPerson, IStarship } from './Board';
import { STARSHIPS_SUIT } from './cardSuits';

interface CardFieldProps {
  fieldValue: number | string;
  fieldName: string;
}

const CardField: React.FC<CardFieldProps & React.HTMLAttributes<HTMLDivElement>> = ({
  className,
  fieldValue,
  fieldName,
}) => {
  return (
    <div className={className}>
      <label>{fieldName}</label>
      <span>{fieldValue}</span>
    </div>
  );
};

export const StyledCardField = styled(CardField)`
  display: flex;
  justify-content: space-between;

  label {
    font-weight: bold;
  }

  span {
    text-align: end;
  }
`;

export type ICard = IStarship | IPerson;

interface CardProps {
  data: ICard;
}

const Card: React.FC<CardProps & React.HTMLAttributes<HTMLDivElement>> = ({ className, data }) => {
  return (
    <div className={className}>
      <div>
        <img src={STARSHIPS_SUIT['c3RhcnNoaXBzOjI=']} alt='img' />
      </div>
      <h2>{data.name}</h2>
      <div>
        {Object.keys(data)
          .filter((key) => !['__typename', 'id', 'name'].includes(key))
          .map((key) => (
            <StyledCardField
              key={key}
              fieldName={camelCaseToSentenceCase(key)}
              fieldValue={data[key]}
            />
          ))}
      </div>
    </div>
  );
};

export const StyledCard = styled(Card)`
  width: 300px;
  padding: 10px;
  border-radius: 10px;
  background: ${colors.white};
  img {
    width: 100%;
  }
`;

export default StyledCard;
