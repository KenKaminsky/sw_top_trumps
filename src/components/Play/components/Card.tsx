import React from 'react';
import { useParams } from 'react-router-dom';
import { Entity } from '../../../apollo_client/types';
import { RouterParams } from '../../../shared/types';
import { camelCaseToSentenceCase } from '../../helpers/index';
import { CardBody, CardBodyRow, CardContainer, CardHeader, Label, Span } from '../styles';

export const IDENTITY_FIELDS = ['__typename', 'id', 'name'];

interface CardProps {
  card: Entity;
}

const Card: React.FC<CardProps> = ({ card }) => {
  const { suit } = useParams<RouterParams>();

  return (
    <CardContainer data-testid={`card-test-id-${card.id}`} variant={'secondary'}>
      <CardHeader suit={suit}>{card.name}</CardHeader>
      <CardBody data-testid={`card-body-id-${card.id}`}>
        {Object.entries(card)
          .filter(([key]) => !IDENTITY_FIELDS.includes(key))
          .map(([key, val]) => (
            <CardBodyRow key={key}>
              <Label>{camelCaseToSentenceCase(key)}</Label>
              <Span>{val}</Span>
            </CardBodyRow>
          ))}
      </CardBody>
    </CardContainer>
  );
};

export default Card;
