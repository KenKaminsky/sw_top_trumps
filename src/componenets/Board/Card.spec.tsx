import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';

describe('Card Snapshot Test', () => {
  it('should render as expected', () => {
    const tree = renderer.create(<Card />).toJSON();

    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});

let container = null;
describe('Card Component', () => {
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
});
