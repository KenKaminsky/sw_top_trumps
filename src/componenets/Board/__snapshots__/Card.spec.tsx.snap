// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot Test should render as expected 1`] = `
<div
  className="sc-gsTCUz fSqflo"
>
  <div>
    <img
      alt="img"
      src=""
    />
  </div>
  <div
    className="sc-bdfBwQ dODQao"
  >
    <h2>
      Name!
    </h2>
    <h5>
      Model
    </h5>
  </div>
  <div>
    <h5>
      Length
    </h5>
    <h5>
      Cargo Capacity
    </h5>
    <h5>
      maxAtmospheringSpeed
    </h5>
    <h5>
      hyperdriveRating
    </h5>
  </div>
</div>
`;
